Introduction to Sites API

KateVomBerg opened the pull request.

rwo-trackunit commented:
I'm not much of writer, but here i technical focused list, of site api features:

Site create/update/archive/delete:
- Site has detailed information such as name, area, location and polygon associated with the site
- Once the site is created, you will gain the ability to see on-site assets and their historical data.
- Once deleted, all linked information will be unavailable, including site history and metadata
  - Not sure it is true for history, but thats what it said in description of the method
- Creation 
  - You must provide a link to a suggestion or a polygon
    - Suggestion: Use this to copy polygon from a suggestion. Can be omitted if polygon is set.
    - Polygon: Polygon in GeoJson format. Can be omitted if suggestion is set.
  - If you omit address, it will be filled in based Google maps lookup on the center of the polygon.
  - You cannot provide the types CLASSIC_POI and CLASSIC_ZONE since they are for Classic migration
- Start and end date will affect the status
  - If start date is in the future, status is PLANNED
  - If end date is in the part, status is FINISHED
  - Else the status is ACTIVE
  - Setting the status to ARCHIVED overwrites this logic

Site assets:
- Exposes asset-on-site live data
- Up to date seconds after the location data has been recived.
- This data may be inaccurate, since location data can be delayed and out-of-order.

Site history
- Exposes asset-on-site historical data. 
- This data is updated hourly
- Since location data can be delayed, data may change back in time.

Site Intervals is deprecated version of Site history, and should be removed.

Site suggestions:
- Automatically detected sites based on your assets' locations. The suggested sites reflect locations where assets are or have been previously, as well as those assets type and usage pattern.
- Can be used when creating a site.
- Live data: Changes hourly.

Site contact
- Simple "tagging" of user information related to the site. This is meant for users outside of Trackunit.

Site user
- Users given permission to see the assets of the site, as well as the site itself.
- Users may already have this permissions, being admin or fleet admin.

Site image?
SiteKPI?
SiteInsight?

rwo-trackunit commented on api-docs/sites-api-intro.md:
```diff
@@ -0,0 +1,66 @@
+
+**Define a Site:**
+- Address: define a site by providing the location via street, ZIP code, city and country (Note: if you omit address, it will be filled in based Google maps lookup on the center of the polygon)
+- Polygon: define a site by providing an array of GPS points, where each point consists of two or three coordinates (longitude, latitude, optionally altitude)
```
Maybe we should have something about the standard here?
Polygon follows [GeoJson](https://geojson.org/) standard and there by [rfc7946](https://datatracker.ietf.org/doc/html/rfc7946)

KateVomBerg commented:
I've added all of Roland's suggestions and also ensured that we reference the OpenAPI specs for better developer experience and also added information about how to access emissions data for sites 🙏🏻

What is still unclear now is the following:
- Sites Interval endpoint should be removed or at least deprecated
- what is the value of SitesImage (also marked as deprecated?), SitesKPI, SitesInsights?

From my side, we could publish the intro page in the current state and @abetterway2feel your team could address the points above in time? If you agree, please approve @rwo-trackunit @abetterway2feel ✨🤞🏻

rwo-trackunit reviewed (approved):
This is good enough to publish, in my opinion. 
Any addition/corrections can be added later.

rwo-trackunit commented on api-docs/sites-api-intro.md:
```diff
@@ -0,0 +1,83 @@
+- or the sites visited by a list of assets
+
+### Site contact and Site user endpoints
+- Contact is a simple "tagging" of user information related to the site. This is meant for users who do not have access to Trackunit.
```
This explanation is a bit "clunky", but I couldn't find a good way to explain it.
The use case, is when you arrive at a site, and there is a post-it on the gate of who to call to get in. This aims to be the digital equivalent.
